gdal/ogr/ogrsf_frmts/mitab/mitab_tabfile.cpp

```diff
@@ -2825,3 +2860,40 @@ void TABFile::Dump(FILE *fpOut /*=NULL*/)
 }
 
 #endif // DEBUG
+
+/*
+ * SetTabDescription()
+ */
+CPLErr TABFile::SetTabDescription(const char *pszDescription)
+{
+    if(m_eAccessMode == TABRead)
+    {
+        CPLError(CE_Warning, CPLE_AppDefined, "Description will not save in TAB file in readonly mode.");
+    }
+
+    m_bNeedTABRewrite = TRUE;
+    if(nullptr != pszDescription && CPLStrnlen(pszDescription, 260) > 256)
+    {
+        char pszFixedDescription[256];
+        CPLStrlcpy(pszFixedDescription, pszDescription, 256);
+        CPLDebug("MITAB", "Truncate description to 255 bytes. New description is '%s'",
+                pszFixedDescription);
+        return IMapInfoFile::SetMetadataItem(DESCRIPTION_KEY, pszFixedDescription);
+    }
+
+    CPLDebug("MITAB", "Set description to '%s'", pszDescription);
```

Comment:
I belive %s with a null pointer is undefined behaviour

Reply:
Fixed.